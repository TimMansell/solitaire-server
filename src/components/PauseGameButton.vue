<template>
  <Button
    :is-stacked="isStacked"
    @click="toggleGamePaused"
    :disabled="isGameLoading"
    data-test="pause-game-btn"
  >
    <span v-if="!isGamePaused">Pause</span>
    <span v-if="isGamePaused">Resume</span>
  </Button>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'PauseGameButton',
  components: {
    Button,
  },
  props: {
    isStacked: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    ...mapGetters(['isGamePaused', 'isGameLoading']),
  },
  methods: {
    ...mapActions(['toggleGamePaused']),
  },
};
</script>
