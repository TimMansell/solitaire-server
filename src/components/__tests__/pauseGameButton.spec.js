import { shallowMount } from '@vue/test-utils';
import PauseGameButton from '@/components/PauseGameButton.vue';

describe('PauseGameButton.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(PauseGameButton, {
      computed: {
        isEmptyBoard: () => false,
        isOnline: () => true,
        isGamePaused: () => false,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
