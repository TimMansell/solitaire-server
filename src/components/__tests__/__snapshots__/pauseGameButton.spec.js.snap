// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PauseGameButton.vue matches pause snapshot 1`] = `
<button-stub type="default" isstacked="true" size="md" data-test="pause-game-btn">
  Pause
</button-stub>
`;

exports[`PauseGameButton.vue matches resume snapshot 1`] = `
<button-stub type="default" isstacked="true" size="md" data-test="pause-game-btn">
  Pause
</button-stub>
`;

exports[`PauseGameButton.vue matches snapshot 1`] = `
<button-stub type="default" isstacked="true" size="md" data-test="pause-game-btn">
  Pause
</button-stub>
`;
